Add PCF2131 ultra-low power optimization for 5-year battery life

This patch adds power optimization for the PCF2131 RTC by disabling
temperature compensation (TC_DIS) and 1/100 seconds counter (100TH_S_DIS)
to reduce power consumption from ~770nA to ~400nA.

The optimization is automatically applied during probe for PCF2131 devices.

Signed-off-by: Dynamic Devices <support@dynamicdevices.co.uk>

--- a/drivers/rtc/rtc-pcf2127.c
+++ b/drivers/rtc/rtc-pcf2127.c
@@ -32,6 +32,9 @@
 /* Control register 1 */
 #define PCF2127_REG_CTRL1		0x00
 #define PCF2127_BIT_CTRL1_POR_OVRD		BIT(3)
+/* PCF2131 power optimization bits */
+#define PCF2127_BIT_CTRL1_TC_DIS		BIT(6)
+#define PCF2127_BIT_CTRL1_100TH_S_DIS		BIT(3)
 #define PCF2127_BIT_CTRL1_TSF1			BIT(4)
 #define PCF2127_BIT_CTRL1_STOP			BIT(5)
 /* Control register 2 */
@@ -1235,6 +1238,20 @@
 
 	pcf2127_watchdog_init(dev, pcf2127);
 
+	/* Enable ultra-low power mode for PCF2131 (5-year battery life) */
+	if (pcf2127->cfg->type == PCF2131) {
+		ret = regmap_update_bits(pcf2127->regmap, PCF2127_REG_CTRL1,
+					 PCF2127_BIT_CTRL1_TC_DIS | PCF2127_BIT_CTRL1_100TH_S_DIS,
+					 PCF2127_BIT_CTRL1_TC_DIS | PCF2127_BIT_CTRL1_100TH_S_DIS);
+		if (ret) {
+			dev_err(dev, "failed to enable ultra-low power mode\n");
+		} else {
+			dev_info(dev, "PCF2131 ultra-low power mode enabled (~400nA)\n");
+		}
+		/* Don't fail probe if power optimization fails */
+		ret = 0;
+	}
+
 	/*
 	 * Disable battery low/switch-over timestamp and interrupts.
 	 * Clear battery interrupt flags which can block new trigger events.